app/src/main/java/lt/vilnius/tvarkau/navigation/FragmentTransactionExecutor.kt

```diff
@@ -0,0 +1,56 @@
+package lt.vilnius.tvarkau.navigation
+
+import android.annotation.SuppressLint
+import android.support.v4.app.Fragment
+import android.support.v4.app.FragmentManager
+import lt.vilnius.tvarkau.R
+
+class FragmentTransactionExecutor(private val fragmentManager: FragmentManager) {
+
+    fun replaceWithClearTop(fragment: Fragment) {
+        clearBackStackTop()
+        replace(fragment, false)
+    }
+
+    @SuppressLint("CommitTransaction")
+    fun replace(fragment: Fragment, addToBackStack: Boolean) {
+        fragmentManager.beginTransaction().apply {
```

Comment:
don't use `apply`. It only introduces another nesting level and doesn't help with code readability.
For example this way looks better IMO:
```
fragmentManager.beginTransaction()
    .replace(R.id.container, fragment)
    .setAllowOptimization(true)
    .apply {
        if (addToBackStack) addToBackStack(null)
    }
    .commitAllowingStateLoss()
```